### PR TITLE
Include PR template in PR description if there is just one commit

### LINKED ISSUE #2540: PR description should auto fill when the PR is a single commit
Ext:

image

.com:

image

### PR DESCRIPTION
**Pull request by joshuaobrien**
Fixes #2540

### CODE REVIEW

File: src/github/createPRViewProvider.ts [hunk 1]
```diff
@@ -112,27 +112,40 @@ export class CreatePullRequestViewProvider extends WebviewViewBase implements vs
...
+			const headRepo = this._folderRepositoryManager.findRepo(byRemoteName(this.compareBranch.upstream.remote));
+
+			if (headRepo) {
+				const headBranch = `${headRepo.remote.owner}:${this.compareBranch.name ?? ''}`;
+				const baseBranch = `${this._pullRequestDefaults.owner}:${this._pullRequestDefaults.base}`;
+				const { total_commits } = await origin.compareCommits(baseBranch, headBranch);
+
+				return total_commits;
+			}
+		}
```
Comment (joshuaobrien) [inline src/github/createPRViewProvider.ts:128]: I mostly pulled this out of the body of `getTitle` so as to not duplicate code

File: src/github/createPRViewProvider.ts [hunk 2]
```diff
@@ -112,27 +112,40 @@ export class CreatePullRequestViewProvider extends WebviewViewBase implements vs
...
+
+		return 0;
```
Comment (joshuaobrien) [inline src/github/createPRViewProvider.ts:130]: I wasn't too sure how to handle the case where `this.compareBranch == null`. Advice appreciated!

Comment (alexr00) [reply src/github/createPRViewProvider.ts:137]: When the `upstream` is undefined we can use the git API to get the number of changes. I have added that.

**Review by alexr00 — APPROVED** [review]
Thank you for the PR!

**Comment by glasser** [comment]
For what it's worth, this doesn't seem to be working for me in v0.29.3.  https://github.com/apollographql/apollo-server/pull/5639 is an example of a single-commit PR I just make where I had to put the commit message in the description manually.

**Comment by alexr00** [comment]
We have not yet made out August Milestone release yet (more info on the timing of our releases: https://github.com/microsoft/vscode-pull-request-github/wiki#4-when-will-the-next-release-occur)

**Comment by glasser** [comment]
Ah, OK — I thought that v0.29.3 was after this fix merged and that's what my VSCode is running though.

**Pull request merged**